main.tf

```diff
@@ -231,6 +231,20 @@ resource "google_compute_address" "static_nginx_ip" {
   }
 }
 
+resource "helm_release" "nginx_controller" {
+  for_each         = { for obj in var.nginx_controllers : obj.name => obj }
```

Comment:
@sundarsudandiraraj it should be one nginx-controller per setup, which gets used by all the ingresses in the form of a controller class. Look at this from the perspective of one nginx is running with an IP, for each ingress definition, it actually has a server block defined, and for each path rules in ingress, it has a location block defined. That's how the nginx controller works under the hood. Hence, we will or should never have a use case of multiple nginx controller in the picture.

The setting that this module would be receiving is, more like -

`enable_nginx_controller: true`
`nginx_controller_ip_name: blah`

Nothing more than that. 

To respect the var definition you have created, it should be more like this-

```
variable "nginx_controller" {
  description = "Whether to have nginx controller installed in this cluster setup; with a dedicated IP"
  type = object({
    enabled = bool
    ip_name = string
  })
  default = {}
} 
```

Name and namespace should be always fixed. It's more like a software installation in your cluster, with certain defaults which can not be changed by design. 
`ingress-nginx-controller` should be the name, and `ingress-nginx` should be the namespace.

Reply:
@dibosh Yep i totally understand how we are going to implement and use it. 
Just throwing it out to the community about the option to create multiple nginx controllers in one finds the need for it, as the doc mentioned the idea of multiple ingress nginx controller. https://kubernetes.github.io/ingress-nginx/user-guide/multiple-ingress/#multiple-ingress-nginx-controllers

Anyway, we could go simpler with your suggestion, I have no concerns. One thing to note is the name will be `ingress-nginx` as the helm-release will append the `-controller` while creation. And probably will need to provide a default value of the the type specific with the keys.

Reply:
That multiple controller use case is different, if we ever need to support that, we might be having another var block for `internal_nginx_controller`. So, yeah let's proceed with the single setup.

Reply:
Already pushed, proceed to review please.